Miscellaneous updates to liquidity baking spec

sskeirik opened the pull request:
Updates include:

- fix entrypoint syntax
- add `#XtzBurn` macro for representing burn fee calculations
- add missing burn txns for conversion entrypoints
- add kast script for lb specs and `kmich` support
- mark unused variables to quiet warnings

EDIT: In my history, this is on top of #308, but technically, they don't depend on each other at all, so this could be merged first, if that would be more helpful. I am hopeful that, after this latest round of fixes, the updated Dexter specs should pass.

nishantjr commented on tests/proofs/liquidity-baking/lb-spec.md:
```diff
@@ -71,10 +71,10 @@ We have one case for when `#ceildiv` results in an upwards rounding, and one for
         <nonce> #Nonce(Nonce => Nonce +Int 2) </nonce>
         <knownaddrs> KnownAddresses </knownaddrs>
-        <paramtype> #Type(#LiquidityBakingParamType()) </paramtype>
+        <paramtype> %default |-> #Type(#LiquidityBakingParamType()) </paramtype>
```
Can we initialize this map (for all entrypoints) in `#runProof`? That way we do not need to repeat ourselves for each claim.

nishantjr replied:
Oh, I notice only `%default` has been set up. Don't we need the other entry points too?

sskeirik replied:
Currently, we need a `%default` entrypoint in order to supply the proper semantics of the `SELF` instruction. However, I realized that all of the parameter type maps were initialized incorrectly leading to proof errors (this incorrect initialization also occurs in the Dexter contract so I will fix it there as well).

nishantjr commented on tests/proofs/liquidity-baking/lb-compiled.md:
````diff
@@ -26,7 +26,6 @@ Since we work with specific code that contains a finite number of annotations, w
 ```k
   syntax FieldAnnotation ::= "%deadline"
-                           | "%default"
                            | "%minTokensBought"
                            | "%mintOrBurn"
                            | "%quantity"
````
Use `[token]` and drop the macros:

```suggestion
                           | "%minTokensBought" [token]
                           | "%mintOrBurn"           [token]
                           | "%quantity"                 [token]
```

Don't know why I can suggest for the whole block.

sskeirik replied:
I replaced the macros with token annotations as you asked.

nishantjr reviewed (changes requested):
Looks good besides the changes suggested.

sskeirik commented:
So, this PR has slowly grown in time as I have fixed several issues. One thing that I changed in Dexter that I want to change in LB is I renamed `#XtzBought` to `#CurrencyBought` since it is used when purchasing tokens or when purchasing XTZ.

As I was doing the token-to-xtz proof, I found I made a mistake in the local entrypoints syntax, which should now be fixed in Dexter in LB.

nishantjr commented:
> So, this PR has slowly grown in time as I have fixed several issues. One thing that I changed in Dexter that I want to change in LB is I renamed `#XtzBought` to `#CurrencyBought` since it is used when purchasing tokens or when purchasing XTZ.

Lets limit the scope of this PR. This should only set up the infrastructure for LB so that we can start working on claims in parallel.
I'm OK with merging this after one test claim is working.

nishantjr reviewed (approved):
LGTM
